### PR TITLE
feat(ckb | examples): Add RGBPP spore methods and examples

### PR DESCRIPTION
**Pull request by duanyytop**
## Main Changes

- Create cluster with RGB++ lock
- Create multiple clustered spores with RGB++ lock
- Add create cluster and spores examples

## Tips

- The `cobuild` of cluster and spore includes `toLock` which should be updated when BTC tx is ready
- The size of the `cobuild` increases linearly with the number of spores. When the number of spores is 200, the `cobuild` size is approximately 59k bytes. And the `cobuild` witness size will affect tx fee.  Please @ShookLyngs take a look

### CODE REVIEW

**Comment by Flouse** [comment]
Please rebase on develop branch to fix the failed test.

File: packages/ckb/package.json [hunk 1]
```diff
@@ -18,18 +18,20 @@
...
-    "@ckb-lumos/base": "0.21.1",
-    "@ckb-lumos/codec": "0.22.0-next.5",
+    "@ckb-lumos/base": "^0.21.1",
+    "@ckb-lumos/codec": "^0.22.1",
+    "@spore-sdk/core": "^0.2.0-beta.6",
     "@nervosnetwork/ckb-sdk-core": "^0.109.1",
     "@nervosnetwork/ckb-sdk-utils": "^0.109.1",
     "@nervosnetwork/ckb-types": "^0.109.1",
     "@rgbpp-sdk/service": "workspace:^",
+    "@exact-realty/multipart-parser": "^1.0.13",
```
Comment (Flouse) [inline packages/ckb/package.json:28]: This package is not used?

Comment (duanyytop) [reply packages/ckb/package.json:28]: The `lumos`  SDK and `spore`  SDK need the `@exact-realty/multipart-parser`, without it, they will occur error.

see: Nervape Tech telegram group context

File: packages/ckb/src/spore/cluster.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+}: CreateClusterCkbVirtualTxParams): Promise<SporeVirtualTxResult> => {
+  const rgbppLock = {
+    ...getRgbppLockScript(isMainnet),
+    args: append0x(rgbppLockArgs),
+  };
+  const rgbppCells = await collector.getCells({ lock: rgbppLock });
+  if (!rgbppCells || rgbppCells.length === 0) {
+    throw new NoRgbppLiveCellError('No rgbpp cells found with the rgbpp lock args');
+  }
+  const rgbppCell = rgbppCells[0];
```
Comment (Flouse) [inline packages/ckb/src/spore/cluster.ts:43]: Only unlocking rgbppCells[0] might not be enough.
related issue:
- https://github.com/ckb-cell/rgbpp-sdk/issues/66
- https://github.com/ckb-cell/rgbpp-sdk/issues/23

I think we could refactor it in another PR, what do you think? @duanyytop

Comment (duanyytop) [reply packages/ckb/src/spore/cluster.ts:43]: Agree.

My initial design was to simplify the creation of clusters and spores and lower the threshold for use by application teams. A universal design is indeed needed.

**Review by Flouse — APPROVED** [review]

**Comment by Dawn-githup** [comment]
https://www.notion.so/RGBPP-spore-methods-and-examples-test-bdd96e07809141ee96123ce9b145eb8c

**Pull request merged**